zk-token-sdk/src/encryption/auth_encryption.rs

```diff
@@ -210,6 +210,31 @@ impl SeedDerivable for AeKey {
     }
 }
 
+impl From<[u8; AE_KEY_LEN]> for AeKey {
+    fn from(bytes: [u8; AE_KEY_LEN]) -> Self {
+        Self(bytes)
+    }
+}
+
+impl From<AeKey> for [u8; AE_KEY_LEN] {
+    fn from(key: AeKey) -> Self {
+        key.0
+    }
+}
+
+impl TryFrom<&[u8]> for AeKey {
+    type Error = AuthenticatedEncryptionError;
+    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
+        if bytes.len() != AE_KEY_LEN {
+            return Err(AuthenticatedEncryptionError::Deserialization);
+        }
+        let Ok(sized_array) = bytes.try_into() else {
+            return Err(AuthenticatedEncryptionError::Deserialization);
+        };
+        Ok(Self(sized_array))
```

Comment:
```suggestion
        bytes
            .try_into()
            .map(Self)
            .map_err(|_| AuthenticatedEncryptionError::Deserialization)
```
Nit, should we make the code slightly more concise as above?